ethstats/ethstats.go

```diff
@@ -583,18 +691,31 @@ func (s *Service) sendStats(conn *websocket.Conn, action string, stats interface
 	//				&& address == addressHasher.digest('hex').substr(24)
 	//				&& msgHash == msgHasher.digest('hex')
 
-	proof := map[string]interface{}{
-		"signature": hexutil.Encode(signature),
-		"address":   etherBase,
-		"publicKey": hexutil.Encode(pubkeyBytes),
-		"msgHash":   msgHash.Hex(),
-	}
-
 	signedStats := map[string]interface{}{
 		"stats": stats,
 		"proof": proof,
 	}
 
+	return signedStats, nil
+}
+
+func (s *Service) sendStats(conn *websocket.Conn, action string, stats interface{}) error {
+	if s.backend.ProxiedPeer() != nil {
+		statsWithAction := map[string]interface{}{
+			"stats":  stats,
+			"action": action,
+		}
+		msg, _ := json.Marshal(statsWithAction)
+		go s.backend.ProxiedPeer().Send(0x15, msg)
```

Comment:
instead of calling `s.backend.ProxiedPeer()` here and before in the if condition, we can pull it out into a variable. I also don't think we should hardcode the literal `0x15`